### DISCUSSION TITLE
How to Handle or Prevent Video/Image Content from status@broadcast Being Stored in MinIO?

### DISCUSSION THREAD

**Discussion by bobykurniawan11** [devlikeapro/waha#716] [Q&A]
I need to implement functionality to manage the deletion or prevent the storage of video and image content associated with status@broadcast in MinIO. This feature is essential to optimize storage usage by removing or blocking obsolete or unnecessary content.

Additionally, I'm looking for a solution to prevent video/image content originating from status@broadcast from being uploaded to the MinIO bucket in the first place. This would reduce storage overhead and eliminate the need for post-upload cleanup.

Current Setup:
```
WAHA_S3_REGION=us-east-1
WAHA_S3_BUCKET=dindwa
WAHA_S3_ACCESS_KEY_ID=aaa
WAHA_S3_SECRET_ACCESS_KEY=xxx
WAHA_S3_ENDPOINT=https://host.host
WAHA_S3_FORCE_PATH_STYLE=True
WAHA_S3_VERSION=latest
```

[![patron:PLUS](https://img.shields.io/badge/patron-PLUS-a0e6ba)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by devlikepro** [comment]
Hi! That's a good question, actually :thinking: 

Even right now you'll have to search `status@broadcast` in `X-Amz-Meta-Waha-Message-Id` S3 metadata, which is not super efficient... 
We need to add `X-Amz-Meta-Waha-Chat-Id` as well, so you can faster lookup for the S3 files associated with the specific chat. We'll add it soon :crossed_fingers: https://github.com/devlikeapro/waha/issues/717

> I'm looking for a solution to prevent video/image content originating from status@broadcast from being uploaded to the MinIO bucket in the first place

If we add a way to ignore all messages from specif `chatId` (like `WAHA_IGNORE_CHATS=status@broadcast` env variable) would it help? Or you need to have text messages from status, just don't want to store the media files?

Another option would be to have `WAHA_LAZY_MEDIA=true` which will never download files and you'll need to explicitly ask to do that by requesting [Get Message By Id](https://waha.devlike.pro/docs/how-to/receive-messages/#get-message-by-id) with `downloadMedia=True` flag.
The `message` webhook would still have the flag indicates that the message have the media, you just will need to explicitly ask to download media for that message 
Mayy be we could also add `message.media` webhook, so you know when the media is saved in S3 :thinking: 

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by bobykurniawan11** [comment]
`WAHA_IGNORE_CHATS=status@broadcast` sounds perfect for what i need. It’ll keep media from status@broadcast out of the MinIO bucket, which is exactly what i looking for.

[![patron:PLUS](https://img.shields.io/badge/patron-PLUS-a0e6ba)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)
